spacepy/pycdf/__init__.py

```diff
@@ -3106,8 +3106,13 @@ def __getitem__(self, key):
         result = hslice.create_array()
         if hslice.counts[0] != 0:
             hslice.select()
-            lib.call(const.GET_, const.zVAR_HYPERDATA_,
-                     result.ctypes.data_as(ctypes.c_void_p))
+            if self.sparse_records != const.NO_SPARSERECORDS:
+                lib.call(const.GET_, const.zVAR_HYPERDATA_,
+                         result.ctypes.data_as(ctypes.c_void_p),
+                         ignore=(const.VIRTUAL_RECORD_DATA,))
+            else:
+                lib.call(const.GET_, const.zVAR_HYPERDATA_,
+                         result.ctypes.data_as(ctypes.c_void_p))
```

Comment:
My bad, I did not test it properly in non-sparse mode. I see your point about not supressing CDF warnings, though this particular one seems a bit silly to me. I guess this warning could still be useful when reading a file without knowing it contains sparse variable.
I don't mind if we revert this change.

Reply:
My goal of always passing along warnings from C is a bit funny because the Python and C conventions on warnings/exceptions are different...the C library interface will happily let you ignore warnings (and errors) unless you explicitly handle them, whereas in Python if you don't explicitly handle them they'll make noise. Given that a warning can be suppressed, but a warning never issued can't be retrieved, I would like to go ahead and revert this.

What would make sense (and I'll add it to the punch list) would be to document the warnings handling and how to suppress it. I use a custom warning type specifically to make that easy.

Reply:
I made the warning documentation a separate issue (#509) rather than lump it into here.